common/src/main/java/org/ergoplatform/appkit/Bip32Serialization.java

```diff
@@ -0,0 +1,95 @@
+                                                         NetworkType networkType) {
+        if (masterKey.path().depth() > 1) {
+            throw new IllegalArgumentException("Master key expected for serialization");
+        }
+
+        DerivationPath eip3Path = Constants.eip3DerivationPath();
+        ExtendedPublicKey eip3Parent = ((ExtendedSecretKey) masterKey.derive(
+            new DerivationPath((Seq) eip3Path.decodedPath().dropRight(1),
+                masterKey.path().publicBranch()))).publicKey();
+        // we need the parent's parent for its fingerprint
+        ExtendedPublicKey eip3ParentParent = ((ExtendedSecretKey) masterKey.derive(new DerivationPath((Seq) eip3Path.decodedPath().dropRight(2),
+            masterKey.path().publicBranch()))).publicKey();
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        out.write(getPublicHeader(networkType), 0, 4);
+        out.write((byte) (eip3Parent.path().depth() - 1));
+        out.write(calculateFingerPrint(eip3ParentParent.keyBytes()), 0, 4);
+        out.write(new byte[]{0, 0, 0, 0}, 0, 4);
+        out.write(eip3Parent.chainCode(), 0, 32);
+        out.write(eip3Parent.keyBytes(), 0, 33);
+        return out.toByteArray();
+    }
+
+    private static byte[] calculateFingerPrint(byte[] key) {
+        byte[] hashedKey = Sha256.hash(key);
+        return Arrays.copyOfRange(new RIPEMD160.Digest().digest(hashedKey), 0, 4);
+    }
+
+    private static byte[] getPublicHeader(NetworkType networkType) {
+        if (networkType == NetworkType.MAINNET) {
+            return new byte[]{0x04, (byte) 0x88, (byte) 0xB2, 0x1E};
+        } else {
+            return new byte[]{0x04, 0x35, (byte) 0x87, (byte) 0xCF};
+        }
+    }
+
+    /**
+     * parses a hex-encoded serialized public key and returns it to use for address derivation
+     */
+    public static ExtendedPublicKey parseExtendedPublicKeyFromHex(String hexEncodedKey, NetworkType networkType) {
+        byte[] xpub = Base16$.MODULE$.decode(hexEncodedKey).get();
+        return parseExtendedPublicKey(xpub, networkType);
+    }
+
+    /**
+     * parses a serialized public key and returns it to use for address derivation
+     */
+    public static ExtendedPublicKey parseExtendedPublicKey(byte[] serializedKey, NetworkType networkType) {
+        if (serializedKey.length != 78) {
+            throw new IllegalArgumentException("xpubkey should have length of 78 bytes.");
+        }
+        if (!Arrays.equals(getPublicHeader(networkType), Arrays.copyOfRange(serializedKey, 0, 4))) {
+            throw new IllegalArgumentException("Given xpubkey did not start with expected version bytes.");
+        }
+
+        return new ExtendedPublicKey(Arrays.copyOfRange(serializedKey, 45, 78),
+            Arrays.copyOfRange(serializedKey, 13, 32 + 13),
+            new DerivationPath((Seq) Constants.eip3DerivationPath().decodedPath().dropRight(1),
+                true));
+    }
+}
```

Comment:
Serializaition roundtrip tests are necessary. See for example https://github.com/ScorexFoundation/sigmastate-interpreter/blob/9e1467ce0a4e981e64246aaeffc35a2efe053978/sigmastate/src/test/scala/org/ergoplatform/ErgoAddressSpecification.scala#L29